icepyx/core/visualization.py

```diff
@@ -312,7 +312,13 @@ def make_request(self, base_url, payload):
         --------
         request_OA_data
         """
-        return requests.get(base_url, params=payload)
+        response = requests.get(base_url, params=payload)
+        if not response.ok:
+            raise RuntimeError(
+                f"Status {response.status_code} requesting url {response.request.url}"
+            )
```

Comment:
We're currently getting a 404 from the API, and later on we try to access the "data" key from the response. Instead of a KeyError at that time, we'll now error at the point of failure.

Reply:
> We're currently getting a 404 from the API

Good to know - I haven't been tracking how the response has changed as things have on the API side.

> Instead of a KeyError at that time, we'll now error at the point of failure.

Awesome!